picasso/src/main/java/com/squareup/picasso3/SourceBufferingInputStream.java

```diff
@@ -33,33 +35,49 @@
   private final Buffer buffer;
   private long position;
   private long mark = -1;
+  private long markLimit = -1;
 
   SourceBufferingInputStream(BufferedSource source) {
     this.source = source;
     this.buffer = source.buffer();
   }
 
   private final Buffer temp = new Buffer();
+  // offset is the write offset in the dest array
   private int copyTo(byte[] sink, int offset, int byteCount) {
     // TODO replace this with https://github.com/square/okio/issues/362
-    buffer.copyTo(temp, offset, byteCount);
```

Comment:
`buffer.copyTo()` treats `offset` as the position in the source buffer to read from, whereas `buffer.read()` treats it as the offset in the output buffer to write to.